src/Symfony/Component/ExpressionLanguage/Parser.php

```diff
@@ -85,21 +86,25 @@ public function __construct(array $functions)
      * variable 'container' can be used in the expression
      * but the compiled code will use 'this'.
      *
-     * @return Node\Node A node tree
+     * In "dry run" mode, the syntax of the passed expression will be checked, but not parsed.
+     * In this case allowed null for names for skip checking dynamic values.
+     *
+     * @return Node\Node|null A node tree or null when called on "dry run" mode
      *
      * @throws SyntaxError
      */
-    public function parse(TokenStream $stream, array $names = [])
+    public function parse(TokenStream $stream, ?array $names = [], bool $dryRun = false)
```

Comment:
Changing the signature this way is a BC break.
Shouldn't we add a new method instead?

Reply:
Yes you are right. Done. Thanks!